gen/cpp/sym/factors/internal_imu_factor.h

```diff
@@ -0,0 +1,2415 @@
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+#include <sym/rot3.h>
+
+namespace sym {
+
+/**
+ * An internal helper function to calculate a sort of between factor from the orientation,
+ * velocity, and position at one time step and at another time step given, where the expected
+ * difference is calculated from the preintegrated IMU measurements between those time steps
+ *
+ * NOTE: If you are looking for a residual for an IMU factor, do not use this. Instead use
+ * the one found in symforce/slam/imu_preintegration/imu_factor.h.
+ *
+ * Time step i is the time of the first IMU measurement of the interval.
+ * Time step j is the time after the last IMU measurement of the interval.
+ *
+ * Precondition: sqrt_info is upper triangular
+ *
+ * Args:
+ *     pose_i (sf.Rot3): Pose at time step i
+ *     vel_i (sf.V3): Velocity at time step i
+ *     pose_j (sf.Rot3): Pose at time step j
+ *     vel_j (sf.V3): Velocity at time step j
+ *     accel_bias_i (sf.V3): The bias of the accelerometer measurements between timesteps i and j
+ *     gyro_bias_i (sf.V3): The bias of the gyroscope measurements between timesteps i and j
+ *     DR (sf.Rot3): Preintegrated estimate for pose_i.R.inverse() * pose_j.R
+ *     Dv (sf.V3): Preintegrated estimate for vel_j - vel_i in the body frame at timestep i
+ *     Dp (sf.V3): Preintegrated estimate for pos change (before velocity and gravity corrections)
+ *         in the body frame at timestep i: R_i^T (p_j - p_i - v_i Delta t - 1/2 g Delta t^2),
+ *         where R_i = pose_i.R, p_i = pose_i.t, and p_j = pose_j.t
+ *     sqrt_info (sf.M99): sqrt info matrix of DR('s tangent space), Dv, Dp
+ *     DR_D_gyro_bias (sf.M33): Derivative of DR w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dp_D_accel_bias (sf.M33): Derivative of Dp w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dp_D_gyro_bias (sf.M33): Derivative of Dp w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dv_D_accel_bias (sf.M33): Derivative of Dv w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dv_D_gyro_bias (sf.M33): Derivative of Dv w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     accel_bias_hat (sf.V3): The bias of the accelerometer measurements used during
+ * pre-integration gyro_bias_hat (sf.V3): The bias of the gyroscope measurements used during
+ * pre-integration gravity (sf.V3): Acceleration due to gravity (in the same frame as pose_x and
+ * vel_x) dt (T.Scalar): The time between timestep i and j: t_j - t_i epsilon (T.Scalar): epsilon
+ * used for numerical stability jacobian: (9x24) jacobian of res wrt args pose_i (6), vel_i (3),
+ * pose_j (6), vel_j (3), accel_bias_i (3), gyro_bias_i (3) hessian: (24x24) Gauss-Newton hessian
+ * for args pose_i (6), vel_i (3), pose_j (6), vel_j (3), accel_bias_i (3), gyro_bias_i (3) rhs:
+ * (24x1) Gauss-Newton rhs for args pose_i (6), vel_i (3), pose_j (6), vel_j (3), accel_bias_i (3),
+ * gyro_bias_i (3)
```

Comment:
Okay. I just lightly looked into it. I think this might be a bug in our docstring wrapping. I noticed that in the doc-string of the symbolic python function, the line for `accel_bias_hat` argument was 102 characters long (over the 100 character limit). I think this might break whatever code formats the doc-string.

When I broke the offending line across two separate lines, the formatting became normal again.

Note: this problem was only present in the generated C++ code (and not the generated python code).

Reply:
> Note: this problem was only present in the generated C++ code (and not the generated python code).

Ah, I think clang-format will wrap comment lines longer than the line length, but our black version/config won't, that's probably why